packages/block-library/src/block/edit.js

```diff
@@ -45,8 +64,39 @@ export default function ReusableBlockEdit( { attributes: { ref } } ) {
 		ref
 	);
 
+	useEffect( () => {
+		// We only track the initial alignment so that if the user could
+		// initially apply wide or full alignments to inner blocks, that ability
+		// is maintained even if they temporarily remove the last inner block
+		// with that wide/full alignment.
+		if ( ! blocks?.length || inferredAlignment !== undefined ) {
+			return;
+		}
+
+		const isConstrained = __unstableParentLayout.type === 'constrained';
```

Comment:
`__unstableParentLayout` can (theoretically) be `undefined` ([here](https://github.com/WordPress/gutenberg/blob/fec05ee8451ee1a8da43cdbf00485736e2b0a16a/packages/block-editor/src/components/block-list/block.js#L138)), though in practice it _should_ always fall back to `default` layout type. Just to be safe, shall we add in optional chaining here?

```suggestion
		const isConstrained = __unstableParentLayout?.type === 'constrained';
```

Reply:
Good call, thanks.